dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/GenericFilter.java

```diff
@@ -125,7 +136,9 @@ public Result invoke(Invoker<?> invoker, Invocation inv) throws RpcException {
                     }
                 } else if (ProtocolUtils.isBeanGenericSerialization(generic)) {
                     return new RpcResult(JavaBeanSerializeUtil.serialize(result.getValue(), JavaBeanAccessor.METHOD));
-                } else {
+                } else if(ProtocolUtils.isProtobufGenericSerialization(generic)) {
```

Comment:
Even move ProtobufUtils into dubbo-rpc-api, it is still a little bit heavier to let dubbo-rpc-api to depend on Protobuf. 

At least we need to file a new issue to track down this work, to refactor this filter to use extension loader to load generic serializations to avoid introduce unnecessary dependencies in this module.